SetGame/SetGame/SetGameModel.swift

```diff
@@ -0,0 +1,274 @@
+//
+//  SetGameModel.swift
+//  SetGame
+//
+//  Created by Nogah Melamed Cohen on 10/10/2019.
+//  Copyright © 2019 Nogah Melamed Cohen. All rights reserved.
+//
+
+import Foundation
+
+struct SetGameModel {
+    var stackCards: [Card]
+    var cardOnBoard: [Card]
+    var selectedCards: [Card]
+    var score = 0
+    var choosenCardsState = CardState.chosen
+
+    var countCardsOnBoard: Int {
+        get {
+            var res = 0
+            for card in self.cardOnBoard {
+                if card != nil {
+                    res += 1
+                }
+            }
+            return res
+        }
+    }
+    
+    init() {
+        self.stackCards = [Card]()
+        self.cardOnBoard = [Card]()
+        self.selectedCards = [Card]()
+        createStack()
+        fillBoard()
+    }
+    
+    //create 81 cards for stack
+    mutating func createStack() {
+        let propertyValues = CardProperty.allCases
+        for color in propertyValues {
+            for shape in propertyValues {
+                for filling in propertyValues {
+                    for shapeCount in propertyValues {
+                        let currCard = Card(shape: shape, color: color, shapeCount: shapeCount, filling: filling)
+                        self.stackCards.append(currCard)
+                    }
+                }
+            }
+        }
+    }
+    
+    //add 12 cards from stack to board
+    mutating func fillBoard() {
+        for index in 0..<12 {
+            self.cardOnBoard.append(self.stackCards.randomElement()!)
+            self.stackCards.remove(element: self.cardOnBoard[index])
+        }
+    }
+    
+    mutating func cardSelected(cardIndex: Int) {
```

Comment:
I love the way you write the code in this method (and in other places of course). The code is clean and the comments make it really easy to understand, good work 💪